zap/src/main/java/org/parosproxy/paros/view/MainMenuBar.java

```diff
@@ -321,6 +326,46 @@ public void actionPerformed(java.awt.event.ActionEvent e) {
         return menuFileOpen;
     }
 
+    private JMenu getMenuFileOpenRecent() {
+        if (menuFileOpenRecent == null) {
+            menuFileOpenRecent = new javax.swing.JMenu();
+            menuFileOpenRecent.setText(Constant.messages.getString("menu.file.openRecent"));
+            for (JMenuItem menuItem : getMenuFileOpenRecentSessions()) {
+                menuFileOpenRecent.add(menuItem);
+            }
+        }
+        return menuFileOpenRecent;
+    }
+
+    public void setMenuFileOpenRecent() {
+        menuFileOpenRecent.removeAll();
+        recentSessionList = null;
+        for (JMenuItem menuItem : getMenuFileOpenRecentSessions()) {
+            menuFileOpenRecent.add(menuItem);
+        }
+    }
+
+    private ArrayList<ZapMenuItem> getMenuFileOpenRecentSessions() {
+        if (recentSessionList == null) {
+            recentSessionList = new ArrayList<>();
+            for (String path : Control.getRecentSessionList()) {
+                ZapMenuItem tmpJMenuItem = new ZapMenuItem(path, path, null);
+                tmpJMenuItem.addActionListener(
+                        new java.awt.event.ActionListener() {
+                            @Override
+                            public void actionPerformed(java.awt.event.ActionEvent e) {
+                                getMenuFileControl().openSession(path);
+                                if (!Model.getSingleton().getSession().isNewState()) {
+                                    toggleSnapshotState(true);
+                                }
+                            }
+                        });
```

Comment:
This can be simplified with a lambda:
```java
                tmpJMenuItem.addActionListener(e -> {
                                getMenuFileControl().openSession(path);
                                if (!Model.getSingleton().getSession().isNewState()) {
                                    toggleSnapshotState(true);
                                }
                            });
```